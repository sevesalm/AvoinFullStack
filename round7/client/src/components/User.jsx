import React from 'react';

const User = ({ user }) => (
  <div>
    <h1>{user.name}</h1>
    <h2>Added blogs</h2>
    <ul>
      {user.blogs.map(blog => (
        <li key={blog._id}>
          {blog.title} by {blog.author}
        </li>
      ))}
    </ul>
  </div>
);

export default User;
