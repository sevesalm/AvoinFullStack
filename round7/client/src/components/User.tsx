import React from 'react';
import { NavLink } from 'react-router-dom';

import { List, ListItem, ListItemText } from '@material-ui/core';
import { IUser } from '../reducers/userReducer';

const User = ({ user }: { user: IUser }) => (
  <div className="user">
    <h1>{user.name}</h1>
    <h2>Added blogs</h2>
    <List component="nav">
      {user.blogs.map(blog => (
        <ListItem
          key={blog._id}
          button={true}
          component={({ innerRef, ...props }) => (
            <NavLink {...props} to={`/blogs/${blog._id}`} />
          )}
        >
          <ListItemText primary={`${blog.title} by ${blog.author}`} />
        </ListItem>
      ))}
    </List>
  </div>
);

export default User;
