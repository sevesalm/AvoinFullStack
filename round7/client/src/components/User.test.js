import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import User from './User';

describe('User', () => {
  const testUser = {
    blogs: [
      {
        _id: 12345,
        author: 'Author',
        title: 'Title'
      }
    ],
    name: 'Test User'
  };

  it('shows user details', () => {
    const user = mount(
      <MemoryRouter>
        <User user={testUser} />
      </MemoryRouter>
    );
    const userElement = user.find('.user').text();
    expect(userElement).toContain(testUser.name);
    const blogs = user.find('ListItem');
    expect(blogs.length).toEqual(1);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <User user={testUser} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
