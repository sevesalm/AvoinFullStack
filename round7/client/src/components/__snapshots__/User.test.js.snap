// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User renders correctly 1`] = `
<div
  className="user"
>
  <h1>
    Test User
  </h1>
  <h2>
    Added blogs
  </h2>
  <nav
    className="MuiList-root-1 MuiList-padding-2"
  >
    <a
      aria-current={null}
      className="MuiButtonBase-root-15 MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-12 MuiListItem-button-13"
      href="/blogs/12345"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      style={undefined}
      tabIndex="0"
    >
      <div
        className="MuiListItemText-root-18"
      >
        <span
          className="MuiTypography-root-24 MuiTypography-subheading-31 MuiListItemText-primary-21"
        >
          Title by Author
        </span>
      </div>
      <span
        className="MuiTouchRipple-root-49"
      />
    </a>
  </nav>
</div>
`;
